sdk/eventhub/event-hubs/README.md

```diff
@@ -306,36 +306,56 @@ const { EventHubConsumerClient } = require("@azure/event-hubs");
 const { ContainerClient } = require("@azure/storage-blob");
 const { BlobCheckpointStore } = require("@azure/eventhubs-checkpointstore-blob");
 
+const storageAccountConnectionString = "storage-account-connection-string";
+const containerName = "container-name";
+const eventHubConnectionString = "eventhub-connection-string";
+const consumerGroup = "my-consumer-group";
+const eventHubName = "eventHubName";
+
 async function main() {
-  const blobContainerClient = new ContainerClient("storage-connection-string", "container-name");
-  await blobContainerClient.create(); // This can be skipped if the container already exists
+  const blobContainerClient = new ContainerClient(storageAccountConnectionString, containerName);
+
+  if (!(await blobContainerClient.exists())) {
+    await blobContainerClient.create();
+  }
+
   const checkpointStore = new BlobCheckpointStore(blobContainerClient);
   const consumerClient = new EventHubConsumerClient(
-    "my-consumer-group",
-    "connectionString",
-    "eventHubName",
+    consumerGroup,
+    eventHubConnectionString,
+    eventHubName,
     checkpointStore
   );
 
   const subscription = consumerClient.subscribe({
     processEvents: async (events, context) => {
       // event processing code goes here
+      if (events.length === 0) {
+        // If the wait time expires (configured via options in maxWaitTimeInSeconds) Event Hubs
+        // will pass you an empty array.
+        return;
+      }
 
-      // Mark the last event in the batch as processed, so that when the program is restarted
-      // or when the partition is picked up by another process, it can start from the next event
+      // Checkpointing will allow your service to restart and pick
+      // up from where it left off.
```

Comment:
Nit: reword to avoid confusion regarding checkpointing allowing restarts

```suggestion
      // Checkpointing will allow your service to pick up from
      // where it left off when restarting.
```

Same in a few other files